### PR TITLE
Material All Properties templates

### PR DESCRIPTION
**Pull request by benblack75**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,21 @@ O3DE (Open 3D Engine) is an open-source, real-time, multi-platform 3D engine tha
 ## Contribute
```
Comment (HogJonny-AMZN) [inline README.md:5]: Looks like this file might be unintentional in your commit?
I think you need to kick this file from you PR, replace with updates in our read me <dccsi>\readme.md?

I love the ideas of readme with notes, impactful changes, and such.
This file is however the root readme.md for o3de:
https://github.com/o3de/o3de/blob/development/README.md

I think we can just follow the pattern, but log our own readme's:
1. when is is worth altering the core o3de readme?
2. should we just extend the <dccsi>\readme?
3. per-package shared readme at root of this API package <dccsi>\azpy\readme.md?
4. or nested per-module nested <dccsi>\azpy\o3de\readme.md files?

**Review by HogJonny-AMZN — CHANGES_REQUESTED** [review]
left notes

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/o3de/allProperties_enhancedPBR.material [hunk 2]
```diff
@@ -0,0 +1,182 @@
+{
```
[the remark this answers is not in the document]
Comment (benblack75) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/o3de/allProperties_enhancedPBR.material:1]: I generated these by ticking on all possible variations within each material. I had to change each value in order for the parameter to surface in the .material file saved output, but the values were then subsequently reset to defaults within the material itself with a few exceptions. All texture file path parameters are just left blank- the activation of these instances (and the resulting additional parameters) begins with a file path. If they are empty, the parameter will not activate anything in material handling which is exactly what we want- even if there is a default texture file (i.e. the material checkerboard template texture on new materials). The other thing I did is for some of the checkboxes the defaults are set to off- I put them to on by default in the event that a developer forgets to activate them. The reason for this is that some parameters are gated by them, so they will need to be active for all parameters to be reachable. I'll revise to conform to Alt1, but yes, we can always refactor along the way if needed. Thanks!

File: README.md (hunk 1 above)
Comment (benblack75) [reply README.md:5]: I did not create this file- it was included automatically. I assumed it was part of some sort of templated system?

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/o3de/allProperties_enhancedPBR.material (hunk 2 above)
[the remark this answers is not in the document]
Comment (HogJonny-AMZN) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/o3de/allProperties_enhancedPBR.material:1]: yeah makes sense, it's ok for our template to not match the actual default settings - it's easier for for us/them to opt out later (and uncheck features)

**Review by shawstar — COMMENTED** [review]
Looks Good.

**Pull request closed without merging**